### PR TITLE
Using SEG mgmt network for finding out host overlap

### PR DESCRIPTION
**Pull request by akshayhavile**
This PR works on following two issues:
1. [Introducing networkUUID parameter in values.yaml and aviinfrasetting]
2. [Using SEG mgmt network for finding out host overlap]
3. Also update AKO version to 1.10.3

TODO: UT (Working on it)
TODO: FT

### CODE REVIEW

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Comment by akshayhavile** [comment]
skip jira-id-check

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

File: internal/k8s/crdcontroller.go [hunk 1]
```diff
@@ -837,8 +839,10 @@ func checkRefOnController(key, refKey, refValue string) error {
 	uri := fmt.Sprintf("/api/%s?name=%s&fields=name,type,labels,created_by", refModelMap[refKey], refValue)
 
 	// For public clouds, check using network UUID in AWS, normal network API for GCP, skip altogether for Azure.
-	if lib.IsPublicCloud() && refModelMap[refKey] == "network" {
-		if lib.UsesNetworkRef() {
+	// If reference key is network uuid , then check using UUID.
+	// TODO: Arif has to check this logic.
```
Comment (swathinsankaran) [inline internal/k8s/crdcontroller.go:843]: @aaha97 could you please check this?

**Comment by akshayhavile** [comment]
> LGTM.
> 
> Are we planning to include the UTs as part of the same PR?

I will raise different PR for that

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Comment by akshayhavile** [comment]
skip jira-id-check

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Comment by akshayhavile** [comment]
skip jira-id-check

**Comment by akshayhavile** [comment]
build ako

File: internal/k8s/ako_init.go [hunk 2]
```diff
@@ -863,64 +863,7 @@ func (c *AviController) FullSyncK8s(sync bool) error {
...
-			meta, err := meta.Accessor(podObj)
-			if err == nil {
-				resVer := meta.GetResourceVersion()
-				objects.SharedResourceVerInstanceLister().Save(key, resVer)
-			}
-			nodes.DequeueIngestion(key, true)
-		}
-	}
-
+	// Re-order informer loading. all crds- then objects depends upon it.
```
Comment (swathinsankaran) [inline internal/k8s/ako_init.go:866]: should we take care of this in the 1.11.1 release and not in a patch?

File: internal/cache/controller_obj_cache.go
Comment (swathinsankaran) [inline internal/cache/controller_obj_cache.go:1]: I think we have to revist the function `validateNetworkNames` in this file. In it, we are iterating through the VIP network list.

File: internal/k8s/ako_init.go (hunk 2 above)
Comment (akshayhavile) [reply internal/k8s/ako_init.go:866]: Re-ordering is require as Aviinfrasetting can be associated with L4. Vip and node network needs to be populated before L4 processing else those fields will be empty and will result in failure.

File: internal/cache/controller_obj_cache.go
Comment (akshayhavile) [reply internal/cache/controller_obj_cache.go:1]: @swathinsankaran: Thanks. I missed this. We need to introduce uuid specific fix in this function while fetching vrfContext

File: internal/k8s/ako_init.go (hunk 2 above)
Comment (swathinsankaran) [reply internal/k8s/ako_init.go:866]: sure, got it.

File: internal/cache/controller_obj_cache.go
Comment (swathinsankaran) [reply internal/cache/controller_obj_cache.go:1]: adding `PopulateVipNetworkwithUUID` also to the above list.

**Comment by akshayhavile** [comment]
skip jira-id-check

File: internal/cache/controller_obj_cache.go
Comment (akshayhavile) [reply internal/cache/controller_obj_cache.go:1]: Addressed `validateNetworkNames` and `checkAndSetVRFFromNetwork` with uuid logic into it. For `PopulateVipNetworkwithUUID` is not required.

**Review by swathinsankaran — APPROVED** [review]
LGTM

**Comment by swathinsankaran** [comment]
build ako

**Pull request merged**